### ISSUE TITLE
mocking failure doesnt work for google?

### ISSUE THREAD

**Issue by sanjaygir** [omniauth/omniauth#942]
I am trying with google and invalid credentials doesnt work. It successfully authenticates from google account. It works on facebook but not google. For google should i do something different? I appreciate any help! Thanks!

```

 it "can handle authentication error" do
    OmniAuth.config.mock_auth[:google] = :invalid_credentials
    visit '/'
    page.should have_content("Sign in with Twitter")
    click_link "Sign in"
    page.should have_content('Authentication failed.') #this fails
  end
```

**Comment by franee** [comment]
Have the same issue, auth_hash is set to the :invalid_credentials and still goes to the callback_path. It doesn't trigger the configured on_failure proc.

Facebook provider works fine btw.

**Comment by sanjaygir** [comment]
please find solution here. https://stackoverflow.com/questions/52169390/how-to-mock-failure-for-google-oauth

**Comment by franee** [comment]
Nope, it was different for me as I was trying to make the callback path be dynamic. Works fine on on development but was acting weird on specs. It was the mock calls that were failing due to on_path? in omniauth not returning correctly.
